src/registry/customerProfile/index.stories.tsx

```diff
@@ -158,6 +160,45 @@ export const OpenPreferencesModalWithoutPortalUrl: Story = {
   },
 };
 
+export const WithUpdatePreferencesModalEnabledSetToFalse: Story = {
+  args: {
+    componentDefinition: {
+      id: 'customerProfile',
+      type: 'customerProfile',
+      key: 'customerProfile',
+      label: 'Profile',
+      description:
+        'The "update preferences" button should never be visible because `updatePreferencesModalEnabled` is set to `false`.',
+      digitalAddressTypes: ['email'],
+      shouldUpdateCustomerData: false,
+    },
+  },
+  parameters: {
+    formSettings: {
+      componentParameters: {
+        customerProfile: {
+          fetchDigitalAddresses: async () => [],
+          portalUrl: '',
+          updatePreferencesModalEnabled: false,
+        },
+      } satisfies FormSettings['componentParameters'],
+    },
+  },
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+    const emailField = await canvas.findByLabelText('Email');
+
+    // Enter an email address
+    await userEvent.type(emailField, 'test@mail.com');
+    emailField.blur();
+
+    // The "update preferences" button should not be visible
+    await waitFor(() => {
+      expect(canvas.queryByRole('button', {name: 'Update preferences'})).not.toBeInTheDocument();
+    });
```

Comment:
These assertions are hard to get right - the assert can pass for many different (wrong) reasons:

* the button is there, but with a different label
* the component is async and momentarily doesn't show it, but does show it after 500ms for whatever reason

So I don't trust these kind of assertions and tend to leave them out myself, instead relying on a good story name, comment in the test and visual check in Chromatic that no unexpected elements are there.

Long story to suggest removing this assertion because IMO it's too fragile.